src/identities/identities.controller.ts

```diff
@@ -622,4 +623,58 @@ export class IdentitiesController {
 
     return new GroupedInstructionModel(result);
   }
+
+  @ApiOperation({
+    summary: 'Rotate Primary Key',
+    description:
+      'Creates an Authorization to rotate primary key of the signing Identity by the `targetAccount`.',
+  })
+  @ApiTransactionResponse({
+    description: 'Newly created Authorization Request along with transaction details',
+    type: CreatedAuthorizationRequestModel,
+  })
+  @ApiBadRequestResponse({
+    description:
+      'The target Account already has a pending invitation to become the primary key of the given Identity',
+  })
+  @Post('rotate-primary-key')
+  async rotatePrimaryKey(
+    @Body() rotatePrimaryKeyDto: RotatePrimaryKeyParamsDto
+  ): Promise<TransactionResponseModel> {
+    const serviceResult = await this.identitiesService.rotatePrimaryKey(rotatePrimaryKeyDto);
+
+    return handleServiceResult(serviceResult, authorizationRequestResolver);
+  }
+
+  @ApiOperation({
+    summary: 'Attest Primary Key Rotation',
+    description: 'Creates an Authorization to attest the primary key rotation for Identity',
+  })
+  @ApiTransactionResponse({
+    description: 'Newly created Authorization Request along with transaction details',
+    type: CreatedAuthorizationRequestModel,
+  })
+  @ApiBadRequestResponse({
+    description:
+      'The target Account already has a pending attestation to become the primary key of the target Identity',
```

Comment:
```suggestion
      'The target Account does not have a pending attestation to become the primary key of the target Identity',
```

Looks like updated copy paste here